src/lib/common/mtmodule.py

```diff
@@ -1,17 +1,71 @@
 from abc import ABC, abstractmethod
-from lib.common.util import save_logs
-from lib.common.exceptions import ImproperLoggedPhaseError
+from lib.common.util import save_logs, hashdict
+from lib.common.exceptions import ImproperLoggedPhaseError, BatchedPhaseArgNotGenerator
 from lib.common.etypes import Etype
 from functools import partial, wraps
+from types import GeneratorType
+from itertools import islice, chain
 import os
+import multiprocessing
+import struct
 
+MAX_CPUS = multiprocessing.cpu_count() - 1
+MIN_ELEMENTS_PER_CPU = 4
```

Comment:
I think we worked out that, even in the case where there are _less_ elements than there are CPUs, we still want to parallelise the separate `analyse_element` function, right? 

This was code I wrote, but I think we should drop this, and change the heuristic of `get_batch_size` to simply return the min of `MAX_CPUS` and `ls_len`. Sound good?